apps/setting/models_provider/impl/azure_model_provider/model/embedding.py

```diff
@@ -0,0 +1,25 @@
+# coding=utf-8
+"""
+    @project: MaxKB
+    @Author：虎
+    @file： embedding.py
+    @date：2024/7/12 17:44
+    @desc:
+"""
+from typing import Dict
+
+from langchain_openai import AzureOpenAIEmbeddings
+
+from setting.models_provider.base_model_provider import MaxKBBaseModel
+
+
+class AzureOpenAIEmbeddingModel(MaxKBBaseModel, AzureOpenAIEmbeddings):
+    @staticmethod
+    def new_instance(model_type, model_name, model_credential: Dict[str, object], **model_kwargs):
+        return AzureOpenAIEmbeddings(
+            model=model_name,
+            openai_api_key=model_credential.get('api_key'),
+            azure_endpoint=model_credential.get('api_base'),
+            openai_api_version=model_credential.get('api_version'),
+            openai_api_type="azure",
+        )
```

Comment:
The provided code looks generally correct and follows PEP 8 guidelines. Here are some minor recommendations:

1. **Typing Hint**: Ensure that you have imported `Dict` from the `typing` module to use it correctly (`from typing import Dict`).

2. **Class Name**: The class name `AzureOpenAIEmbeddingModel` can be improved for clarity. You might consider using a more descriptive name like `EmbeddedQA`, which aligns with its intended purpose.

3. **Static Method Documentation**: Document what the `new_instance` method does, as static methods typically describe functions without an instance of the class.

   ```python
   @staticmethod
   def new_instance(model_type: str, model_name: str, model_credential: dict, **model_kwargs) -> 'MaxKBBaseModel':
       """Creates a new instance of AzureOpenAIEmbeddingsModel."""
   ```

4. **Variable Naming Conventions**: Keep naming conventions consistent. Use lowercase letters separated by underscores for variable names in Python, unless there's a specific reason not to do so.

5. **Docstring**: Add a docstring at the beginning of the file or class if necessary.

```python
# coding=utf-8
"""
    @project: MaxKB
    @Author：赵虎
    @file： embedding.py
    @date：2024/7/12 17:44
    @desc:
"""

import inspect

from typing import Dict

from langchain_openai import AzureOpenAIEmbeddings

from setting.models_provider.base_model_provider import MaxKBBaseModel


@staticmethod
def new_instance(model_type: str, model_name: str, model_credential: Dict[str, object], **model_kwargs) -> MaxKBBaseModel:
    """
    Creates a new instance of EmbeddedQA based on the provided configuration.

    Parameters:
    - model_type (str): Type of the model.
    - model_name (str): Name of the model.
    - model_credential (dict): Credential dictionary containing API key and endpoint information.
    - model_kwargs (kwargs): Additional parameters to configure the model.

    Returns:
    - MaxKBBaseModel: A newly created instance of EmbeddedQA.
    """
    return AzureOpenAIEmbeddings(
        model=model_name,
        openai_api_key=model_credential.get('api_key', None),
        azure_endpoint=model_credential.get('api_base', None),
        openai_api_version=model_credential.get('api_version', "2023-06"),
        openai_api_type="azure",
    )
```

This improves readability, makes the function self-explanatory, and enhances maintainability.